client/src/components/deleteModal.js

```diff
@@ -0,0 +1,62 @@
+import { useState } from 'react';
+import Button from 'react-bootstrap/Button';
+import Modal from 'react-bootstrap/Modal';
+
+function DeleteModal({id}) {
+  const [show, setShow] = useState(false);
+
+  const handleClose = () => setShow(false);
+  const handleShow = () => setShow(true);
+
+  /**
+         * Deletes a record from the database and updates the state of the records.
+         * @param {string} id - The ID of the record to be deleted.
+         * @returns {Promise<void>} A promise that resolves when the record is deleted.
+         */
+  async function deleteRecord(id) {
+    await fetch(`http://localhost:5000/${id}`, {
+        method: "DELETE"});
+    }
+
+  return (
+    <>
+    {/* {console.log(typeof(id))}
+    {console.log(id)} */}
+      <button className="btn btn-link" onClick={handleShow}>Delete</button>
+
+      <Modal
+        show={show}
+        onHide={handleClose}
+        backdrop="static"
+        keyboard={false}
+      >
+        <Modal.Header closeButton>
+          <Modal.Title>Delete Record?</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          Are you sure that you want to delete this record?
+        </Modal.Body>
+        <Modal.Footer>
+          <Button variant="secondary" onClick={handleClose}>
+            Close
+          </Button>
+          <Button variant="primary" onClick={() => {
+            deleteRecord(id);
+            sleep(500).then(() => { window.location.reload(); });
```

Comment:
I don't think  sleep is necessary here. `window.location.reload();` alone should be okay.

Reply:
I tested with removing this, the page does not update without it.